### PR TITLE
Update details of BBE : Objects can now refer non-abstract objects

### PR DESCRIPTION
**Pull request by KRVPerera**
## Purpose
Objects can now include non-abstract objects. This PR updates description.

Refer spec change: https://github.com/ballerina-platform/ballerina-lang/issues/23359

May need to improve the BBEs too.

### CODE REVIEW

File: examples/object-type-reference/object_type_reference.bal [hunk 1]
```diff
@@ -38,8 +38,11 @@ type Manager object {
 
     public string dpt;
 
+    // referenced fields can be overridden by subtypes
+    public float salary;
```
Comment (ThisaruGuruge) [inline examples/object-type-reference/object_type_reference.bal:42]: If the salary is a `float`, the `init` function should assign a `float` value to the `salary` field, right?

Comment (KRVPerera) [reply examples/object-type-reference/object_type_reference.bal:42]: Fixed : a0fb6d56bd8b7475a57a4228932fc8ba58c28396

File: examples/object-type-reference/object_type_reference.bal [hunk 2]
```diff
@@ -38,13 +38,16 @@ type Manager object {
 
     public string dpt;
 
+    // referenced fields can be overridden by subtypes
```
Comment (MaryamZi) [inline examples/object-type-reference/object_type_reference.bal:41]:
```suggestion
    // Referenced fields can be overridden by sub-types given that the type of the field  
    // in the sub-type is a sub-type of the original type of the field.
```

File: examples/object-type-reference/object_type_reference.bal [hunk 3]
```diff
@@ -53,6 +56,7 @@ type Manager object {
         return self.firstName + " " + self.lastName;
     }
 
+    // refered methods can be overridden with a subtype method
```
Comment (MaryamZi) [inline examples/object-type-reference/object_type_reference.bal:59]:
```suggestion
    // Reference methods can also be overridden as long the method in the sub-type is 
    // a sub-type of the method in the referenced type.
```

File: examples/object-type-reference/object_type_reference.description [hunk 4]
```diff
@@ -1,4 +1,6 @@
-// Object type references provide a way to copy the members from an abstract 
-// object into another object. It is equivalent to specifying the members 
-// explicitly within the new object, thus, eliminating redundancy. An object can have
+// Object type references provide a way to copy the members from an object 
+// into another object. It is equivalent to specifying the members explicitly
+// within the new object, thus, eliminating redundancy. An object can have
 // zero or more type references and may have chained references as well.
+// Fields and method definitions can override referenced fields and methods
+// provided that they are subtypes of referenced types.
```
Comment (MaryamZi) [inline examples/object-type-reference/object_type_reference.description:6]:
```suggestion
// provided that they are sub-types of referenced field/method types.
```

File: examples/object-type-reference/object_type_reference.bal [hunk 5]
```diff
@@ -53,6 +57,8 @@ type Manager object {
         return self.firstName + " " + self.lastName;
     }
 
+    // Referenced methods can also be overridden as long the method in the overriding
```
Comment (MaryamZi) [inline examples/object-type-reference/object_type_reference.bal:60]:
```suggestion
    // Referenced methods can also be overridden as long as the method in the overriding 
```

**Pull request merged**